plugins/chain/plugin.cpp

```diff
@@ -348,8 +353,10 @@ namespace chain {
             ilog("Opening shared memory from ${path}", ("path", my->shared_memory_dir.generic_string()));
             my->db.open(data_dir, my->shared_memory_dir, STEEMIT_INIT_SUPPLY, my->shared_memory_size, chainbase::database::read_write/*, my->validate_invariants*/ );
             auto head_block_log = my->db.get_block_log().head();
-            my->replay |= head_block_log && my->db.revision() != head_block_log->block_num();
-
+            if (my->replay_if_struct_changed) {
+                my->replay |= head_block_log && my->db.revision() != head_block_log->block_num();
+            }
```

Comment:
I just was not know what it should quit, I just read: it should do not replay. Not added to catch because catch is only replay and no more (if do not quit). 

If this place will not be reached at all, I should remove 355-362 lines, yes?

Reply:
Yes, the result of corrupted shared memory (changed structure is one of the corrupted shared memory) is an exception from `database::open()`. So lines 355-358 will never reached.